api/PayrollProcessor.Core.Domain/Intrastructure/Operations/Commands/ICommand.cs

```diff
@@ -0,0 +1,6 @@
+namespace PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands
+{
+    public interface ICommand { }
```

Comment:
What's nice about baking the `TryOptionAsync<T>` into the `ICommandHandler` is that I don't need to expose it in the `ICommand`.